plugin/test-keeper/plugin/test_matcher_test.go

```diff
@@ -0,0 +1,137 @@
+			verifyThatMatcherMatchesCorrectFile(GoMatcher, "/path/test/test_anything.go", false)
+			verifyThatMatcherMatchesCorrectFile(GoMatcher, "/path/test/anytest.go", false)
+
+			// when test file then true
+			verifyThatMatcherMatchesCorrectFile(GoMatcher, "/path/to/my_test.go", true)
+		})
+
+		It("JavaScriptMatcher should return true only when matches JavaScript test file", func() {
+			// when non test file then false
+			verifyThatMatcherMatchesCorrectFile(JavaScriptMatcher, "/path/to/my.assertion.js", false)
+			verifyThatMatcherMatchesCorrectFile(JavaScriptMatcher, "/path/test/my.assertion.js", false)
+			verifyThatMatcherMatchesCorrectFile(JavaScriptMatcher, "/path/test/test.anything.js", false)
+
+			// when test file then true
+			verifyThatMatcherMatchesCorrectFile(JavaScriptMatcher, "/path/test/any.test.js", true)
+			verifyThatMatcherMatchesCorrectFile(JavaScriptMatcher, "/path/to/my.test.js", true)
+			verifyThatMatcherMatchesCorrectFile(JavaScriptMatcher, "/path/to/my.spec.js", true)
+		})
+
+		It("TypeScriptMatcher should return true only when matches TypeScript test file", func() {
+			// when non test file then false
+			verifyThatMatcherMatchesCorrectFile(TypeScriptMatcher, "/path/to/my.assertion.ts", false)
+			verifyThatMatcherMatchesCorrectFile(TypeScriptMatcher, "/path/test/my.assertion.tsx", false)
+			verifyThatMatcherMatchesCorrectFile(TypeScriptMatcher, "/path/test/test.anything.ts", false)
+
+			// when test file then true
+			verifyThatMatcherMatchesCorrectFile(TypeScriptMatcher, "/path/test/any.test.tsx", true)
+			verifyThatMatcherMatchesCorrectFile(TypeScriptMatcher, "/path/to/my.test.ts", true)
+			verifyThatMatcherMatchesCorrectFile(TypeScriptMatcher, "/path/to/my.spec.ts", true)
+		})
+
+		It("PythonMatcher should return true only when matches Python test file", func() {
+			// when non test file then false
+			verifyThatMatcherMatchesCorrectFile(PythonMatcher, "/path/to/my_assertion.py", false)
+			verifyThatMatcherMatchesCorrectFile(PythonMatcher, "/path/test/my_assertion.py", false)
+
+			// when test file then true
+			verifyThatMatcherMatchesCorrectFile(PythonMatcher, "/path/test/test_anything.py", true)
+			verifyThatMatcherMatchesCorrectFile(PythonMatcher, "/path/to/my_test.py", true)
+		})
+
+		It("GroovyMatcher should return true only when matches Groovy test file", func() {
+			// when non test file then false
+			verifyThatMatcherMatchesCorrectFile(GroovyMatcher, "/path/to/MyAssertion.groovy", false)
+			verifyThatMatcherMatchesCorrectFile(GroovyMatcher, "/path/test/MyAssertions.groovy", false)
+
+			// when test file then true
+			verifyThatMatcherMatchesCorrectFile(GroovyMatcher, "/path/test/TestAnything.groovy", true)
+			verifyThatMatcherMatchesCorrectFile(GroovyMatcher, "/path/test/MyTest.groovy", true)
+			verifyThatMatcherMatchesCorrectFile(GroovyMatcher, "/path/test/MyTests.groovy", true)
+			verifyThatMatcherMatchesCorrectFile(GroovyMatcher, "/path/test/MyTestCase.groovy", true)
```

Comment:
Out of curiosity - did you look at https://onsi.github.io/ginkgo/#table-driven-tests?

Reply:
nope, thanks for the tip. 
It won't help much with the code as I still need some method for creating the message, but it will separate every entry for isolated test execution. This means that if there are more failures in one table, then you will see all of them, not only the first one.